frontend/simple/utils/giLodash.js

```diff
@@ -12,3 +12,169 @@ export function mapValues (obj: Object, fn: Function, o: Object = {}) {
 export function cloneDeep (obj: JSONObject) {
   return JSON.parse(JSON.stringify(obj))
 }
+
+function isMergeableObject (val) {
+  var nonNullObject = val && typeof val === 'object'
+  return nonNullObject && Object.prototype.toString.call(val) !== '[object RegExp]' && Object.prototype.toString.call(val) !== '[object Date]'
+}
+
+export function merge (obj: Object, src: Object) {
+  for (let key in src) {
+    if (isMergeableObject(obj[key])) {
+      if (isMergeableObject(src[key])) {
+        let clone = cloneDeep(src[key])
+        merge(obj[key], clone)
+      } else {
+        obj[key] = src[key]
+      }
+    } else {
+      if (isMergeableObject(src[key])) {
+        let clone = cloneDeep(src[key])
+        obj[key] = clone
+      } else {
+        obj[key] = src[key]
+      }
+    }
+  }
+}
+
+export function debounce (func: Function, wait: Number, options: Object) {
+  if (typeof func !== 'function') {
+    throw new TypeError('Invalid Function')
+  }
+  let lastArgs
+  let lastThis
+  let maxWait
+  let result
+  let timerId
+  let lastCallTime
+  let lastInvokeTime = 0
+  let leading = false
+  let trailing = true
+  let maxing = false
+
+  if (!wait && wait !== 0) {
+    wait = 0
+  }
+
+  if (options) {
+    leading = !!options.leading
+    maxing = 'maxWait' in options
+    maxWait = maxing ? Math.max(+options.maxWait || 0, wait) : maxWait
+    trailing = 'trailing' in options ? !!options.trailing : trailing
+  }
+
+  function invokeFunc (time) {
+    const args = lastArgs
+    const thisArg = lastThis
+
+    lastArgs = lastThis = undefined
+    lastInvokeTime = time
+    result = func.apply(thisArg, args)
+    return result
+  }
+
+  function startTimer (pendingFunc, wait) {
+    return setTimeout(pendingFunc, wait)
+  }
+
+  function cancelTimer (id) {
+    clearTimeout(id)
+  }
```

Comment:
What's the advantage of using the methods `startTimer` or `cancelTimer` when they do exactly the same as the native method?

Reply:
I'll do this and more